src/main/java/hudson/plugins/translation/L10nDecorator.java

```diff
@@ -97,9 +100,15 @@ public String encodeRecording(StaplerRequest request) throws IOException {
      * Does the opposite of {@link #encodeRecording(StaplerRequest)}.
      */
     public List<Msg> decode(StaplerRequest request) throws IOException {
-        BufferedReader r = new BufferedReader(new InputStreamReader(new GZIPInputStream(new CipherInputStream(
+        final GZIPInputStream gzipInpurStream = new GZIPInputStream(new CipherInputStream(
                 new ByteArrayInputStream(Base64.decode(request.getParameter("bundles").toCharArray())),
-                getCipher(DECRYPT_MODE)))));
+                getCipher(DECRYPT_MODE)));  
+        final BufferedReader r;
+        try {
+            r = new BufferedReader(new InputStreamReader(gzipInpurStream, "UTF-8"));
+        } catch (UnsupportedEncodingException ex) {
+            throw new IOException("UTF-8 encoding is not supported");
```

Comment:
http://docs.oracle.com/javase/6/docs/api/java/nio/charset/Charset.html says:

> # Standard charsets
>
> Every implementation of the Java platform is required to support the following standard charsets. Consult the release documentation for your implementation to see if any other charsets are supported. The behavior of such optional charsets may differ between implementations.
><table><thead><tr><th>Charset</th><th>Description</th></tr></thead><tbody>
><tr><td>US-ASCII</td><td>Seven-bit ASCII, a.k.a. ISO646-US, a.k.a. the Basic Latin block of the Unicode character set</td></tr>
><tr><td>ISO-8859-1</td><td>ISO Latin Alphabet No. 1, a.k.a. ISO-LATIN-1</td></tr>
><tr><td>UTF-8</td><td>Eight-bit UCS Transformation Format</td></tr>
><tr><td>UTF-16BE</td><td>Sixteen-bit UCS Transformation Format, big-endian byte order</td></tr>
><tr><td>UTF-16LE</td><td>Sixteen-bit UCS Transformation Format, little-endian byte order</td></tr>
><tr><td>UTF-16</td><td>Sixteen-bit UCS Transformation Format, byte order identified by an optional byte-order mark</td></tr></tbody></table>

So if you are running Java 6+ it is a JVM Error to get an `UnsupportedEncodingException` for `UTF-8` and you should treat it as such.

(And before you ask, UTF-8 is in the Java 5 JVM specification too http://docs.oracle.com/javase/1.5.0/docs/api/java/nio/charset/Charset.html)